### PR TITLE
Do not emit value from startWith after immediate (sync) disposal from the consumer

### PR DESCRIPTION
**Pull request by Andarist**
Similar to https://github.com/cyclejs/callbags/pull/13

### CODE REVIEW

File: src/startWith.ts [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
 
 export function startWith<T>(value: T): Operator<T, T> {
   return source => (_, sink) => {
+    let ended = false;
     source(0, (t, d) => {
       if (t === 0) {
-        sink(0, d);
+        sink(0, (_: END) => {
+          ended = true;
+          d(_);
```
Comment (jvanbruegge) [inline src/startWith.ts]: please use a number `2` here

**Comment by jvanbruegge** [comment]
Thanks for catching this!

File: src/startWith.ts (hunk 1 above)
Comment (Andarist) [reply src/startWith.ts]: changed

**Comment by staltz** [comment]
Thanks! Looks good to merge

**Pull request merged**